### PR TITLE
Add patch pull request

### PR DESCRIPTION
**Pull request by tboogh**
This PR implements updating a pull request.

https://docs.github.com/en/rest/pulls/pulls#update-a-pull-request

### CODE REVIEW

File: OctoKit/PullRequest.swift [hunk 1]
```diff
@@ -233,12 +309,27 @@ enum PullRequestRouter: JSONPostRouter {
                 parameters["head"] = head
             }
 
+            return parameters
+        case let .patchPullRequest(_, _, _, _, title, body, state, base, mantainerCanModify):
+            var parameters = [
+                "title": title,
+                "state": state.rawValue, // FIXME: All is invalid here?
```
Comment (pietbrauer) [inline OctoKit/PullRequest.swift:316]: Is this still valid? I saw you added a guard let earlier which is fine by me.

Comment (tboogh) [reply OctoKit/PullRequest.swift:316]: Missed removing it, I'll push an update removing the fix

**Comment by tboogh** [comment]
@pietbrauer I pushed an update to fix the lint error

**Review by pietbrauer — APPROVED** [review]

**Comment by pietbrauer** [comment]
Thanks a lot!

**Pull request merged**